src/Propel/Common/Config/PropelConfiguration.php

```diff
@@ -29,7 +29,7 @@ class PropelConfiguration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
+        $treeBuilder = new TreeBuilder(); //FIXME! invoked with 0 parameters, 1-3 required
```

Comment:
This only belongs to `symfony/config >= 4.2` 

See the version in 4.1:
https://github.com/symfony/config/blob/4.1/Definition/Builder/TreeBuilder.php

Reply:
We'll need a new issue created for symfony/config >= 4.2, and we should introduce a constraint to keep it below 4.2 until we have a fix.

Reply:
Some switch then, indeed.

Reply:
Yes but I think we can remove the comment and add a new issue for it. Perhaps the constraint should get added in this PR however because this is where we find the issue.

Reply:
Just merge https://github.com/propelorm/Propel2/pull/1582 first, and I will rebase on top of that
It will then solve it all nicely

Reply:
We would need to drop sf 2.x and 3.x first or make that fix legacy-bc.

Reply:
What do u propose here? Reverting this Change for now?

Reply:
Yes, let's revert this line

```suggestion
        $treeBuilder = new TreeBuilder();
```

In a new PR: We can either add the BC code or add constraints against symfony/config >= 4.2 if we really need to. I can document it as a new issue.